index.js

```diff
@@ -1,20 +1,55 @@
 'use strict';
-const crypto = require('crypto');
+const {Worker} = require('worker_threads');
 
-const create = algorithm => async (buffer, options) => {
+let worker; // Lazy
+let taskIdCounter = 0;
+const tasks = new Map();
+
+const taskWorker = (value, transferList) => new Promise(resolve => {
+	const id = taskIdCounter++;
+	tasks.set(id, resolve);
+	if (worker === undefined) {
+		worker = new Worker('./thread.js');
+		worker.on('message', msg => {
+			const task = tasks.get(msg.id);
+			tasks.delete(msg.id);
+			if (tasks.size === 0) {
+				worker.unref();
+			}
+
+			task(msg.value);
+		});
+		worker.on('error', err => {
```

Comment:
Generally, good to use `once` in events you know will be singular, to prevent possible memory leaks.

Reply:
It's only created once for thread, so no memory leak possible. Theoretically, errors can be multiple(at least it's not specified if there can only be one), so it may introduce problems with debugging in future.

Reply:
Oh, and we don't recreate the thread after error. (Because if it ever happens we really don't know what to do and in general it means something went _extremely_ wrong.)

Reply:
refactored that part